lib/downloader.rb

```diff
@@ -0,0 +1,78 @@
+require 'io/console'
+require 'net/http'
+require 'uri'
+require_relative 'const'
+require_relative 'color'
+require_relative 'convert_size'
+
+def setTermSize
+  # setTermSize: set progress bar size based on terminal width
+  # get terminal window size
+  @termH, @termW = IO.console.winsize
+  # space for progress bar after minus the reserved space for showing
+  # the file size and progress percentage
+  @progBarW = @termW - 2 - (9 + 1) - (4 + 1)
+  return true
+end
+
+def downloader (url, filename = File.basename(url), retry_count = 0, verbose: false)
+  setTermSize
+  # reset width settings after terminal resized
+  trap('WINCH') { setTermSize }
+  uri = URI(url)
+
+  Net::HTTP.start(uri.host, uri.port, {
+      max_retries: 3,
+      use_ssl: uri.scheme.eql?('https'),
+      ca_file: SSL_CERT_FILE,
+      ca_path: SSL_CERT_DIR
+  }) do |http|
+    http.request( Net::HTTP::Get.new(uri) ) do |response|
+      case
+      when response.is_a?(Net::HTTPOK)
+      when response.is_a?(Net::HTTPRedirection) # follow HTTP redirection
+        puts <<~EOT if verbose
+          * Follow HTTP redirection: #{response['Location']}
+          *
+        EOT
+        return downloader(response['Location'], filename, retry_count, verbose: verbose)
```

Comment:
`verbose = false`?

Reply:
Actually, you can just chop off `: verbose` since `verbose` already has a value at this point.

Reply:
Done